src/Momento.Sdk/Responses/CacheKeysExistResponse.cs

```diff
@@ -0,0 +1,118 @@
+﻿namespace Momento.Sdk.Responses;
+
+using System.Collections.Generic;
+using System.Linq;
+using Google.Protobuf;
+using Momento.Protos.CacheClient;
+using Momento.Sdk.Exceptions;
+
+/// <summary>
+/// Parent response type for a cache KeysExist request. The
+/// response object is resolved to a type-safe object of one of
+/// the following subtypes:
+/// <list type="bullet">
+/// <item><description>CacheKeysExistResponse.Success</description></item>
+/// <item><description>CacheKeysExistResponse.Error</description></item>
+/// </list>
+/// Pattern matching can be used to operate on the appropriate subtype.
+/// For example:
+/// <code>
+/// if (response is CacheKeysExistResponse.Success successResponse)
+/// {
+///     // handle success as appropriate
+/// }
+/// else if (response is CacheKeysExistResponse.Error errorResponse)
+/// {
+///     // handle error as appropriate
+/// }
+/// else
+/// {
+///     // handle unexpected response
+/// }
+/// </code>
+/// </summary>
+public abstract class CacheKeysExistResponse
+{
+
+    /// <include file="../docs.xml" path='docs/class[@name="Success"]/description/*' />
+    public class Success : CacheKeysExistResponse
+    {
+        private readonly List<ByteString> keys;
+
+        /// <summary>
+        /// An ordered collection of bools representing whether each key in the original
+        /// request exists in the cache or not.  The order is the same as the order that
+        /// the keys were specified in.
+        /// </summary>
+        public IEnumerable<bool> ExistsEnumerable { get; private set; }
+
+        /// <summary>
+        /// A dictionary whose keys are the cache keys from the request, and whose values are
+        ///  True if the specified key exists in the cache, false otherwise.
+        /// </summary>
+        public IDictionary<string, bool> ExistsDictionary
```

Comment:
This is not something we have in the Go SDK.
I'd assume we had a specific .NTE customer request for this?
Even so, should we make it availabe in the Go SDK as well?

Reply:
We can do it on-demand, or whenever we have time.  I was just trying to think of how I would be most likely to want to use this if I was writing code against it, and most likely the first thing I would do with an "array of booleans" type of response is dump it into a Map/Dictionary so that I can then write a conditional around `resultsDict["foo"]` rather than having to correlate my keys with the booleans based on their index.  So it seems like a really convenient way to access the data.

I'm open to considering omitting it but with my "user" hat on, I think this is what I would want.